src/wire/causal_snapshot.rs

```diff
@@ -0,0 +1,200 @@
+//! A wire protocol for representing Modality probe causal snaphots
+
+use crate::{wire::le_bytes, ProbeId};
+use core::mem::size_of;
+use static_assertions::const_assert_eq;
+
+/// Everything that can go wrong when attempting to interpret a causal snaphot
+/// from the wire representation
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum CausalSnapshotWireError {
+    /// There weren't enough bytes for a full causal snapshot
+    MissingBytes,
+    /// The probe id didn't follow the rules for being
+    /// a valid Modality probe-specifying ProbeId
+    InvalidProbeId(u32),
+}
+
+/// A read/write wrapper around a causal snaphot buffer
+#[derive(Debug, Clone)]
+pub struct CausalSnapshot<T: AsRef<[u8]>> {
```

Comment:
Seems odd to have a wire wrapper type with the same name as the canonical in-memory representation type.